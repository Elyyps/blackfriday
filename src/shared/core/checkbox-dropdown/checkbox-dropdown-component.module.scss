.content {
  width: 200px;
}

.counter {
  font-size: 15px;
  color: #777777;
  margin-bottom: 12px;
}

.filter-bar {
  display: flex;
  justify-content: space-between;
  align-items: center;

  cursor: pointer;
}
