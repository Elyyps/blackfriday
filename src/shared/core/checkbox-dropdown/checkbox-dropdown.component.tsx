import * as React from "react";

import styles from "./checkbox-dropdown-component.module.scss";
import { FilterItem } from "@app/api/core/filter/filter-item";
import { CheckboxComponent } from "../checkbox";
import { DropdownComponent } from "../dropdown-new/dropdown.component";

export interface ICheckboxDropDownProps {
  items: FilterItem[];
  onChange?: (items: FilterItem[]) => void;
  title: string;
}

const CheckboxDropDown = (props: ICheckboxDropDownProps) => {
  const onChecked = (value: any) => {
    const result = props.items.map(item => {
      if (item.displayName === value) {
        item.isSelected = !item.isSelected;
      }

      return item;
    });
    if (props.onChange) {
      props.onChange(result);
    }
  };

  return (
    <div>
      <DropdownComponent title={props.title}>
        <React.Fragment>
          {props.items.map((item, key) => (
            <li key={key} className={styles["filter-bar-li"]}>
              <CheckboxComponent isChecked={item.isSelected} onClick={onChecked}>
                {item.displayName}
              </CheckboxComponent>
              <span className={styles["counter"]}>{item.totalAmount}</span>
            </li>
          ))}
        </React.Fragment>
      </DropdownComponent>
    </div>
  );
};

export { CheckboxDropDown };
