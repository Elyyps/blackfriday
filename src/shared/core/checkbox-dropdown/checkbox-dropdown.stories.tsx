import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { CheckboxDropdown } from "./checkbox-dropdown.component";

storiesOf("Checkbox dropdown", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <CheckboxDropdown items={[]} onChange={jest.fn()} title="" />);
