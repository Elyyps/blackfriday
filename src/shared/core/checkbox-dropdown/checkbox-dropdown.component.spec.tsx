import React from "react";
import { shallow } from "enzyme";

import { CheckboxDropdown } from "@app/core/checkbox-dropdown/checkbox-dropdown.component";

describe("[CheckboxDropdown]", () => {
  it("should render without crashing", () => {
    const renderedComponent = shallow(<CheckboxDropdown items={[]} onChange={jest.fn()} title="" />);

    expect(renderedComponent).toMatchSnapshot();
  });
});
