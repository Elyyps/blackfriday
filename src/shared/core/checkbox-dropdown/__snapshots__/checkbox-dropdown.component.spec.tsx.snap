// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[CheckboxDropdown] should render without crashing 1`] = `
<DropdownComponent
  hasSelectedItems={false}
  isOpen={false}
  setIsOpen={[Function]}
  title=""
>
  <div
    className="content"
  />
</DropdownComponent>
`;
